tests/Doctrine/Tests/ORM/Functional/OneToManyMatchingCriteriaTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\Common\Collections\Criteria;
+use Doctrine\ORM\Mapping\FetchMode;
+use Doctrine\Tests\Models\ECommerce\ECommerceCategory;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * Tests sorting mechanism of Selectable::matching()
+ */
+class OneToManyMatchingCriteriaTest extends OrmFunctionalTestCase
+{
+    private $parent;
+    private $firstChild;
+    private $secondChild;
+    private $thirdChild;
+
+    protected function setUp() : void
+    {
+        $this->useModelSet('ecommerce');
```

Comment:
On newer tests we decided to no longer use the model sets, instead we're putting the required entities together in the same file of the test (with only the necessary properties and methods to reproduce the issue).

This helps us to more easily analyse the system under test and find out if the solution is the best. Could you please change this test in that way?

Reply:
Done, added a GH7767ParentEntity and a GH7767ChildEntity.